hs_geographic_feature_resource/receivers.py

```diff
@@ -0,0 +1,522 @@
+                # validate_files_dict['message'] = " The shp, shx or dbf files can not be removed."
+                all_file_removed=True
+                one_file_removed=False
+
+                for f in ResourceFile.objects.filter(object_id=res_obj.id):
+                    res_f_fullname = f.resource_file.name.lower()
+                    res_f_fullname= res_f_fullname[res_f_fullname.rfind('/')+1:]
+                    if res_f_fullname != del_f_fullname:
+                        f.resource_file.delete()
+                        f.delete()
+
+            elif del_f_ext == ".prj":
+                originalcoverage_obj=res_obj.metadata.originalcoverage.all().first()
+                res_obj.metadata.update_element('OriginalCoverage', element_id=originalcoverage_obj.id, projection_string=UNKNOWN_STR, projection_name =UNKNOWN_STR, datum =UNKNOWN_STR, unit =UNKNOWN_STR)
+                res_obj.metadata.coverages.all().delete()
+
+            if one_file_removed:
+                ori_fn_dict=json.loads(ori_file_info.filenameString)
+                if del_f_fullname in ori_fn_dict:
+                    del ori_fn_dict[del_f_fullname]
+                    res_obj.metadata.update_element('OriginalFileInfo', element_id=ori_file_info.id,filenameString=json.dumps(ori_fn_dict))
+            elif all_file_removed:
+                res_obj.metadata.originalfileinfo.all().delete()
+                res_obj.metadata.geometryinformation.all().delete()
+                res_obj.metadata.fieldinformation.all().delete()
+                res_obj.metadata.originalcoverage.all().delete()
+                res_obj.metadata.coverages.all().delete()
+
+    pass
+
+@receiver(pre_add_files_to_resource, sender=GeographicFeatureResource)
+def geofeature_pre_add_files_to_resource(sender, **kwargs):
+    if sender is GeographicFeatureResource:
+        res_obj = kwargs['resource']
+        files = kwargs['files']
+        validate_files_dict = kwargs['validate_files']
+        ori_file_info=res_obj.metadata.originalfileinfo.all().first()
+        some_new_files_added=True
+        # res_file_list=ResourceFile.objects.filter(object_id=res_obj.id)
+        # for res_f in res_file_list:
+        #     n=res_f.resource_file.name
+
+        try:
+            #for crt_f
+            if ori_file_info and len(ResourceFile.objects.filter(object_id=res_obj.id))>0:#just add non-required files
+                crt_f_str=ori_file_info.filenameString.lower()
+                for f in files:
+                    new_f_fullname=f.name.lower()
+                    new_f_name, new_f_ext=os.path.splitext(new_f_fullname)
+
+                    if new_f_ext in  [".shp", ".shx", ".dbf"]:
+                        validate_files_dict['are_files_valid'] = False
+                        validate_files_dict['message'] = "No more shp, shx, dbf files can be added."
+                        some_new_files_added=False
+                        break
```

Comment:
Do these `break` statements mean you only process one file even if there are multiple?